net-libs/nativebiginteger/nativebiginteger-0.9.36-r1.ebuild

```diff
@@ -0,0 +1,70 @@
+# Copyright 2018 Valérian Rousset
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit java-pkg-2 toolchain-funcs
+
+DESCRIPTION='jbigi library used by net-vpn/i2p'
+HOMEPAGE='https://geti2p.net'
+SRC_URI="https://download.i2p2.de/releases/${PV}/i2psource_${PV}.tar.bz2"
+
+LICENSE='public-domain'
+SLOT='0'
+KEYWORDS='~amd64 ~x86'
+
+IUSE='test'
+
+DEPEND='
+	dev-libs/gmp:0
+	virtual/jdk:1.8
+'
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/i2p-${PV}/core"
+
+PATCHES=(
+	"${FILESDIR}/${P}-asmfix.patch"
+)
+
+src_compile() {
+	compile_lib() {
+		local name="${1}"
+		local file="${2}"
+		shift 2
+
+		"$(tc-getCC)" "${@}" ${CFLAGS} $(java-pkg_get-jni-cflags) \
+			${LDFLAGS} -shared -fPIC "-Wl,-soname,lib${name}.so" \
+			"${file}" -o "lib${name}.so" ||
+			die "unable to build ${name}"
+	}
+
+	cd "${S}/c/jbigi/jbigi" && compile_lib jbigi src/jbigi.c -Iinclude -lgmp
+
+	if ( use amd64 || use x86 )
```

Comment:
No reason for subshell here. Also you may want to use `if ...; then` on a single line for consistency.